spec/models/user_spec.rb

```diff
@@ -58,6 +62,30 @@
   end
 
   describe "validations" do
+    context 'when mobile is valid' do
+      it 'be valid' do
+        user.mobile = '0123456789'
+        expect(user).to be_valid
+      end
+
+      it 'reject mobile that has more 10 digits' do
+        user.mobile = '012345678991'
+        expect(user).to be_valid
+      end
+    end
+
+    context 'when mobile not valid' do
+      it 'reject mobile that has characters' do
+        user.mobile = '06866666ab'
+        expect(user).not_to be_valid
+      end
+
+      it 'reject mobile that has less 10 digits' do
+        user.mobile = '012345678'
```

Comment:
shoulda can do it to

Reply:
What do you mean, maybe leave it as is?

Reply:
https://github.com/thoughtbot/shoulda-matchers use this gem for this validations

Reply:
refactor